webpack/assets/javascripts/react_app/components/Layout/components/InstanceTitleViewer.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { Icon, OverlayTrigger, Tooltip } from 'patternfly-react';
+import PropTypes from 'prop-types';
+import NavItem from './NavItem';
+
+const InstanceTitleViewer = ({ title }) => {
+  if (!title) {
+    return null;
+  }
+
+  function truncate(str, n) {
+    let res = str;
+    if (str.length > n) {
+      res = `${str.substr(0, n)} \u2026`;
+    }
+    return res;
+  }
```

Comment:
Couldn't the overflow be fixed by setting `max-width` and some css? We even have `ellipsis` css class that handles it:
https://github.com/theforeman/foreman/blob/develop/app/assets/stylesheets/base.scss#L279-L284 
and in react we have https://github.com/amirfefer/react-ellipsis-with-tooltip which @amirfefer created exactly for these cases

Reply:
I think an icon is a better solution because this is about the min width of the header (after that location, org and user move to the side bar). The space that's left between notifications and locations is pretty small so it would fit a few chars (depends on the username)
![Screenshot from 2020-10-04 14-17-17](https://user-images.githubusercontent.com/30431079/95013956-58153c80-064c-11eb-8922-39f555852d8c.png)

Reply:
we tried a lot to get it working but the space is just to small. The icon + tooltip works pretty good and I like the appearance.

Reply:
No matter what `max-width` is going to be set, we still get [overflow](https://github.com/theforeman/foreman/pull/7791#issuecomment-652932604) of the whole nav menu on the right unless we adjust the breakpoints for collapsible layout to hide that menu.

Reply:
:+1:

Reply:
Oh, I missed the fact it was changed to icon+tooltip. In that case I don't think we need to truncate it at all?

Reply:
As I already said: Well, 30 chars is still a very long text. I guess we should still cut it. Just to make sure, that nobody pasts a book in the text. :-)

I can remove it that it truncate after X chars. Opinion @MariaAga  / @xprazak2 ?

Reply:
![image](https://user-images.githubusercontent.com/25485845/95187598-a270f780-07cb-11eb-8ccf-59149925d073.png)
I will remove the section to truncate the message.